File: bms/go/go-gt/initTrace/initTrace.go
package test

var ()

const ()

func initTrace(n, inc int64, trace *Matrix) {
	var i, j int64
	for i = 0; i < n; i++ {
		for j = 0; j < n; j++ {
			trace.Set(i, j, inc)
		}
	}
}

func (m Matrix) Set(i int64, j int64, v int64) {
	m.A[i*m.N+j] = v
}

type Matrix struct {
	N int64
	A []int64
}

File: translations_only/fail/go/go-gt/initTrace/claude2/CAPR/initTrace.rs


struct Matrix {
    n: i64,
    a: Vec<i64>,
}

fn set(matrix: &mut Matrix, i: i64, j: i64, v: i64) {
    matrix.a[i as usize * matrix.n as usize + j as usize] = v;
}

fn init_trace(n: i64, inc: i64, trace: &mut Matrix) {
    let mut i = 0;
    let mut j = 0;
    while i < n {
        while j < n {
            set(trace, i, j, inc);
            j += 1;
        }
        i += 1;
    }
}

